pkg/network/standard/connection.go

```diff
@@ -377,12 +377,17 @@ func (c *Conn) fill(i int) (err error) {
 	// Circulate reading data so that the node holds enough data
 	for i > 0 {
 		n, err := c.c.Read(c.inputBuffer.write.buf[node.malloc:])
+		if n > 0 {
+			node.malloc += n
+			c.inputBuffer.len += n
+			i -= n
+			if err != nil {
```

Comment:
384-386 equals:
```
return nil
```

Reply:
Not equal. If err == nil, it may enter next loop to fill enough data.